# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
# Commit: 09db50c8f370
# Full Hash: 09db50c8f370b69f186f3e6b7ce6154a984b40b4
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 9
# Description:
#   Bug 1654112 - Pass codec configs by value/reference instead of pointers. r=ng
#   
#   This patch mainly sets the stage for using codec configs with StateMirroring.
#   It also makes the code a bit simpler, and removes the need to handle the nullptr
#   case.
# ==============================================================================

diff -r d1552c620c11 -r 09db50c8f370 dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
--- a/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Wed Aug 11 11:33:21 2021 +0200
+++ b/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Wed Apr 14 17:36:40 2021 +0200
@@ -596,7 +596,7 @@
     TransceiverImpl::UpdateConduitRtpExtmap(
         *conduit, details, MediaSessionConduitLocalDirection::kRecv);
 
-    std::vector<UniquePtr<VideoCodecConfig>> configs;
+    std::vector<VideoCodecConfig> configs;
     nsresult rv = TransceiverImpl::NegotiatedDetailsToVideoCodecConfigs(
         details, &configs);
 
@@ -642,7 +642,7 @@
   if (mJsepTransceiver->mRecvTrack.GetNegotiatedDetails() &&
       mJsepTransceiver->mRecvTrack.GetActive()) {
     const auto& details(*mJsepTransceiver->mRecvTrack.GetNegotiatedDetails());
-    std::vector<UniquePtr<AudioCodecConfig>> configs;
+    std::vector<AudioCodecConfig> configs;
     nsresult rv = TransceiverImpl::NegotiatedDetailsToAudioCodecConfigs(
         details, &configs);
 